### PR TITLE
Fix/bpm

### PR DESCRIPTION
**Pull request by envin3**

### CODE REVIEW

File: src/settings/index.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const settings = {
     github: 'https://github.com/pnetwork-association/ptokens-dapp/',
   },
   api: {
-    bpm: 'https://internal.pnetwork.watch:8082/sync-status?apikey=a1d87144c4d60917b880d9fed94d480829d0893a',
+    bpm: 'https://internal.pnetwork.watch:8082/sync-status',
```
Comment (oliviera9) [inline src/settings/index.js:13]: :+1

File: src/hooks/use-swap-info.js [hunk 2]
```diff
@@ -1,46 +1,52 @@
...
+          fromAsset = getAssetById(from.pTokenId)
+        }
+        // ATM, the API returns untrustworthy estimates for EOS-like chains.
+        // For those chains, assume the sync ETA is 0 if the BPM is > 0
+        // as EOS-like chains are usually very fast.
+        const eosLikeChainIds = [...chainIdToTypeMap]
+          .filter(([_k, _v]) => _v === BlockchainType.EOSIO)
+          .map(([_id]) => _id)
+        if (!fromAsset.isNative) {
+          const selectedBpm = Object.values(bpm).find(
```
Comment (oliviera9) [inline src/hooks/use-swap-info.js:23]: can't we factor out this into a `getBpmInfo(chainId,symbol,isNative)` ?
You could reuse that for both cases and move the `try` inside it.

**Review by oliviera9 — COMMENTED** [review]
I would refactor a lil bit. Then remember to bump version

**Comment by oliviera9** [comment]
If you have time it might be worth adding a test to the new getBpmInfo function with a mocked bpm.

File: src/hooks/use-swap-info.js (hunk 2 above)
Comment (envin3) [reply src/hooks/use-swap-info.js:23]: Yes, but I would keep the `try` on all getETA. Anything happening there should block the dapp, right?

Comment (oliviera9) [reply src/hooks/use-swap-info.js:13]: The function is so simple I wouldn't even use a try.
Anyway feel free to keep it since we need to fix it quickly.

Comment (envin3) [reply src/hooks/use-swap-info.js:13]: With the refactor everything would still be handled by the try so I just changed as you said

File: src/hooks/use-swap-info.js [hunk 3]
```diff
@@ -1,11 +1,42 @@
...
+          )
+          return selectedBpm
+            ? eosLikeChainIds.includes(chainId)
+              ? selectedBpm.bpmMedianHost > 0
+                ? 0
+                : -1
+              : selectedBpm.estimatedHostSyncTime
+            : -2
+        } else {
+          const selectedBpm = Object.values(bpm).find(
```
Comment (oliviera9) [inline src/hooks/use-swap-info.js:24]: I would have factored out the
```
          const selectedBpm = Object.values(bpm).find(
            (_el) => _el.bridgeName.includes(`${symbol.toLowerCase()}-on-`) && _el.nativeChainId === chainId
          )
          return selectedBpm
            ? eosLikeChainIds.includes(chainId)
              ? selectedBpm.bpmMedianNative > 0
                ? 0
                : -1
              : selectedBpm.estimatedNativeSyncTime
            : -2
        }
```
block into a function.

File: src/hooks/use-swap-info.js [hunk 4]
```diff
@@ -1,11 +1,42 @@
...
+        if (!isNative) {
+          const selectedBpm = Object.values(bpm).find(
+            (_el) => _el.bridgeName.includes(`${symbol.toLowerCase()}-on-`) && _el.hostChainId === chainId
+          )
+          return selectedBpm
+            ? eosLikeChainIds.includes(chainId)
+              ? selectedBpm.bpmMedianHost > 0
+                ? 0
+                : -1
+              : selectedBpm.estimatedHostSyncTime
```
Comment (oliviera9) [inline src/hooks/use-swap-info.js:21]:
```suggestion
              : isNative ? selectedBpm.estimatedNativeSyncTime : selectedBpm.estimatedHostSyncTime
```

File: src/hooks/use-swap-info.js [hunk 5]
```diff
@@ -1,11 +1,29 @@
...
+      try {
+        const selectedBpm = Object.values(bpm).find(
+          (_el) => _el.bridgeName.includes(`${symbol.toLowerCase()}-on-`) && _el.nativeChainId === chainId
+        )
+        return selectedBpm
+          ? eosLikeChainIds.includes(chainId)
+            ? selectedBpm.bpmMedianNative > 0
+              ? 0
+              : -1
+            : selectedBpm.estimatedNativeSyncTime
```
Comment (oliviera9) [inline src/hooks/use-swap-info.js]: When do you use estimatedHostSyncTime?

**Review by oliviera9 — APPROVED** [review]
LGTM

**Pull request merged**